lib/scarpe/download.rb

```diff
@@ -1,47 +1,87 @@
 # frozen_string_literal: true
 
-require "net/http"
-require "openssl"
-require "nokogiri"
+class Scarpe
+  class Widget
+    def download(url, method: "GET", save: nil, styles: {}, &block)
+      @block = block
 
-class Scarpe::Widget
-  def download(url)
-    Thread.new do
-      begin
-        uri = URI(url)
-        http = Net::HTTP.new(uri.host, uri.port)
-        http.use_ssl = true
-        http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+      Thread.new do
+        logger = ::Scarpe::Logger.logger("Scarpe::App#download")
+        begin
+          uri = URI(url)
+          port = uri.port || (uri.scheme == "https" ? 443 : 80)
+          http = Net::HTTP.start(uri.host, port, use_ssl: uri.scheme == "https", verify_mode: OpenSSL::SSL::VERIFY_NONE)
 
-        request = Net::HTTP::Get.new(uri.request_uri)
+          request = Net::HTTP.const_get(method.capitalize).new(uri.request_uri)
+          apply_styles(request, styles)
 
-        http.request(request) do |response|
-          case response
-          when Net::HTTPSuccess
-            # content = response.body
+          http.request(request) do |response|
+            handle_response(response, save, styles)
```

Comment:
acc to manual ig we have to directly pass this response and update window with response.body but tis resulting in malformed utf and all so i am doing this..

Reply:
and actually updating window with response.body seems pretty crazy

@para.replace response.body

Reply:
So, what I see in the manual is that the download runs for awhile and then runs the handler that was passed to download:

```
 Shoes.app do
   stack do
     title "Searching Google", size: 16
     @status = para "One moment..."

     # Search Google for 'shoes' and print the HTTP headers
     download "http://www.google.com/search?q=shoes" do |goog|
       @status.text = "Headers: " + goog.response.headers.inspect
     end
   end
 end
```

So in that example, goog.response.body would be the body. In the example they ignore the body and put headers into their para.

Reply:
ah oh ig i misread the below lines as example..

Reply:
hm.. but i cant seem to get headers method frm there but i can get headers this way

goog.response.to_hash.inspect #headers 
goog.response.body #body

Reply:
or .each_header is also available i am not sure why i am getting error like header is not available.

Reply:
removed extract_headers method